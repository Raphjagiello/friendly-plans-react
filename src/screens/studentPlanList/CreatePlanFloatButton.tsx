import React from 'react';
import { FloatingAction } from 'react-native-floating-action';

import { Icon } from 'components';
import { palette } from 'styles';

interface Props {
  onPress: () => void;
}

export class CreatePlanFloatButton extends React.PureComponent<Props> {
  render() {
    return (
      <FloatingAction
        overrideWithAction
        actions={[
          {
            name: 'create-plan',
            icon: <Icon name="add" type="material" color={palette.secondary} size={32} />,
          },
        ]}
        onPressItem={this.props.onPress}
      />
    );
  }
}
