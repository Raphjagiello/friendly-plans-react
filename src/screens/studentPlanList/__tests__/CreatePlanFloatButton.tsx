import { shallow } from 'enzyme';
import React from 'react';
import { CreatePlanFloatButton } from '../CreatePlanFloatButton';

describe('CreatePlanFloatButton', () => {
  const props = {
    onPress: jest.fn(),
  };

  it('renders correctly', () => {
    const createPlanFloatButton = shallow(<CreatePlanFloatButton {...props} />);

    expect(createPlanFloatButton).toMatchSnapshot();
  });
});
