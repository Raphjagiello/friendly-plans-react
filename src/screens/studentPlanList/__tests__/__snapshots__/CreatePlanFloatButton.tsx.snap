// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePlanFloatButton renders correctly 1`] = `
<FloatingAction
  actions={
    Array [
      Object {
        "icon": <Icon
          color="#FE9DCA"
          name="add"
          size={32}
          type="material"
        />,
        "name": "create-plan",
      },
    ]
  }
  actionsPaddingTopBottom={8}
  animated={true}
  buttonSize={56}
  color="#1253bc"
  dismissKeyboardOnPress={false}
  distanceToEdge={30}
  iconColor="#fff"
  iconHeight={15}
  iconWidth={15}
  listenKeyboard={false}
  mainVerticalDistance={0}
  onPressItem={[MockFunction]}
  openOnMount={false}
  overlayColor="rgba(68, 68, 68, 0.6)"
  overrideWithAction={true}
  position="right"
  shadow={Object {}}
  showBackground={true}
  visible={true}
/>
`;
